modules/desktop-notifier/src/lib.rs

```diff
@@ -0,0 +1,106 @@
+use std::{
+    os::unix::process::CommandExt,
+    process::{Command, Stdio},
+};
+
+use anyhow::{Context, Result};
+use pulsar_core::{
+    event::Threat,
+    pdk::{
+        CleanExit, ConfigError, ModuleConfig, ModuleContext, ModuleError, PulsarModule,
+        ShutdownSignal, Version,
+    },
+};
+
+const MODULE_NAME: &str = "desktop-notifier";
+
+pub fn module() -> PulsarModule {
+    PulsarModule::new(MODULE_NAME, Version::new(0, 1, 0), desktop_nitifier_task)
+}
+
+async fn desktop_nitifier_task(
+    ctx: ModuleContext,
+    mut shutdown: ShutdownSignal,
+) -> Result<CleanExit, ModuleError> {
+    let mut receiver = ctx.get_receiver();
+    let mut rx_config = ctx.get_cfg::<Config>();
+    let mut config = rx_config.borrow().clone()?;
+
+    loop {
+        let event = tokio::select! {
+            r = shutdown.recv() => return r,
+            _ = rx_config.changed() => {
+                config = rx_config.borrow().clone()?;
+                continue;
+            }
+            msg = receiver.recv() => msg?,
+        };
+        if let Some(Threat { source, info }) = &event.header().threat {
+            let payload = &event.payload();
+            notify(
+                &config,
+                vec![
+                    format!("Pulsar module {source} identified a threat"),
+                    format!("{info}\n Source event: {payload}"),
+                ],
```

Comment:
better to pass data and then create title and content inside the function, but it's only a matter of style

Reply:
I wanted a low level function which could be used also for other notifications, but I agree formatting the title and content in the main task function feels kind of weird. I've created an intermediate `handle_event` function which calls the low-level one.